dkg/messages.go

```diff
@@ -150,6 +158,55 @@ func (msg *Init) Decode(data []byte) error {
 	return json.Unmarshal(data, msg)
 }
 
+// Reshare triggers the resharing protocol
+type Reshare struct {
+	// ValidatorPK is the the public key to be reshared
+	ValidatorPK types.ValidatorPK
+	// EncryptedShares contains the encrypted shares of the old set
+	EncryptedShares map[types.OperatorID][]byte
```

Comment:
The node should already have that, no need to pass it again

Reply:
The intention of adding this is so that this struct is self sufficient. The information can be fed by node. The protocol is agnostic about whether the node grabs the info from local storage or the smart contract. I think previously we included EncryptedShares so that it can be recovered from the smart contract. If we exclude this, the node then has to get the plaintext share and feed to the resharing protocol.

Reply:
Yeah the issue with it is that you will need to add checks that it matches what the node knows anyhow so just another complication